Composer/packages/server/src/models/connector/csharpBotConnector.ts

```diff
@@ -109,3 +177,26 @@ export class CSharpBotConnector implements IBotConnector {
     });
   };
 }
+
+process.on('SIGINT', () => {
+  console.log('[SIGINT] start graceful shutdown');
+  shutdown();
+  process.exit(1);
+});
+process.on('SIGTERM', () => {
+  console.log('[SIGTERM] start graceful shutdown');
+  shutdown();
+  process.exit(1);
+});
+process.on('SIGQUIT', () => {
+  console.log('[SIGQUIT] start graceful shutdown');
+  shutdown();
+  process.exit(1);
+});
+function shutdown() {
+  if (runtime) {
```

Comment:
I see why this is a module variable now. Let's move this to a static property and make it a map `{ pid: runtime }` to support multiple runtimes in the future.